### PR TITLE
fix: avoid throwing exception during plugin loading if the introspect…

### PR DESCRIPTION
**Pull request by leleueri**
… is failing

 In Access-Management so plugin can't be loaded because the policy is using a class onle present in the Gateway classpath
 As proxy & message attributes have no sens for AM, this step should not prevent the load of the plugin

fixes AM-719

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `2.0.2-AM-719-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/plugin/gravitee-plugin/2.0.2-AM-719-SNAPSHOT/gravitee-plugin-2.0.2-AM-719-SNAPSHOT.zip)

### CODE REVIEW

File: gravitee-plugin-policy/src/main/java/io/gravitee/plugin/policy/internal/PolicyPluginHandler.java [hunk 1]
```diff
@@ -66,14 +66,18 @@ protected PolicyPlugin create(Plugin plugin, Class<?> pluginClass) {
...
+            try {
+                Set<String> proxyPhases = new HashSet<>();
+                if (methodFoundFor(REQUEST_ANNOTATIONS, policyClass)) {
+                    proxyPhases.add("REQUEST");
+                }
+                if (methodFoundFor(RESPONSE_ANNOTATIONS, policyClass)) {
+                    proxyPhases.add("RESPONSE");
+                }
+                entity.manifest().properties().put("proxy", proxyPhases.stream().collect(Collectors.joining(",")));
+            } catch (Exception e) {
```
Comment (benoitgravitee) [inline gravitee-plugin-policy/src/main/java/io/gravitee/plugin/policy/internal/PolicyPluginHandler.java:78]: Please use catch-all type syntax `FooException | BarException e`

File: gravitee-plugin-policy/src/main/java/io/gravitee/plugin/policy/internal/PolicyPluginHandler.java [hunk 2]
```diff
@@ -66,14 +66,18 @@ protected PolicyPlugin create(Plugin plugin, Class<?> pluginClass) {
...
+                logger.debug("Unable to autodetect the execution phases for the plugin {}.", entity.id(), e);
```
Comment (benoitgravitee) [inline gravitee-plugin-policy/src/main/java/io/gravitee/plugin/policy/internal/PolicyPluginHandler.java:79]: I understand the context so you don't log it in warning, maybe add a comment to explain when this happens and why it is only debug logging.

File: gravitee-plugin-policy/src/main/java/io/gravitee/plugin/policy/internal/PolicyPluginHandler.java [hunk 3]
```diff
@@ -66,14 +66,18 @@ protected PolicyPlugin create(Plugin plugin, Class<?> pluginClass) {
 
     private void determineProxyPhases(Class policyClass, PolicyPluginImpl entity) {
         if (!entity.manifest().properties().containsKey("proxy") && !entity.manifest().properties().containsKey("message")) {
-            Set<String> proxyPhases = new HashSet<>();
-            if (methodFoundFor(REQUEST_ANNOTATIONS, policyClass)) {
-                proxyPhases.add("REQUEST");
-            }
-            if (methodFoundFor(RESPONSE_ANNOTATIONS, policyClass)) {
-                proxyPhases.add("RESPONSE");
+            try {
```
Comment (benoitgravitee) [inline gravitee-plugin-policy/src/main/java/io/gravitee/plugin/policy/internal/PolicyPluginHandler.java:69]: Ideally, there is a way to detect that the gateway ( or rest api ???) is running (via Spring), and we don't have to run this for AM...

Comment (leleueri) [reply gravitee-plugin-policy/src/main/java/io/gravitee/plugin/policy/internal/PolicyPluginHandler.java:69]: Currently, there is no such info as far as I know.
The only thing we have is the "applcation" name in the Node but it a string define in each project so in the current state, it is hard to adapt the behaviour for APIM/AM project without introducing project specifuc constants.

**Review by benoitgravitee — APPROVED** [review]

**Pull request merged**